Tests/Engine/ModuleHelp.Tests.ps1

```diff
@@ -63,7 +63,7 @@ Param
 	$RequiredVersion
 )
 
-# #Requires -Module @{ModuleName = 'Pester'; ModuleVersion = '3.4.0'}
+# #Requires -Module @{ModuleName = 'Pester'; ModuleVersion = '4.1.1'}
```

Comment:
Is there a technical reason to upgrade to v4 as part of this PR? I think it might be better to do this as a separate PR?

Reply:
the test fixes are sort of entwined, by doing them all at once I won't need to change the tests again.

Reply:
Ok, I see. Then we also need to update the documented version number of Pester in the test section [here](https://github.com/PowerShell/PSScriptAnalyzer/blob/development/README.md#tests) of the developer guide.